sqlglot/transforms.py

```diff
@@ -133,6 +133,48 @@ def remove_precision_parameterized_types(expression: exp.Expression) -> exp.Expr
     )
 
 
+def decode_to_case(expression: exp.Expression) -> exp.Expression:
+    """
+    Transforms the Snowflake/Oracle/Redshift DECODE function into a CASE expression. Note that NULL
+    needs special treatment in CASE expressions, since we need to explicitly check for them using
+    the `IS NULL` predicate, instead of relying on pattern matching.
+
+    - https://stackoverflow.com/a/3209948/17518317
+    - https://docs.snowflake.com/en/sql-reference/functions/decode
+    """
+    if isinstance(expression, exp.Matches):
+        from sqlglot.optimizer.simplify import simplify
+
+        select = expression.this
+        expressions = expression.expressions
+
+        ifs = []
+        for search, result in zip(expressions[::2], expressions[1::2]):
+            search = simplify(search)
```

Comment:
why do you need simpliify

Reply:
It's not needed per se. The reason I used it is so that we don't produce the `IS NULL` checks for expressions consisting of literals that can be reduced to a single literal value, like arithmetic expressions. No problem with removing this completely -- maybe the overhead is not worth it.

Reply:
let’s remove it for now